lib/std/os.zig

```diff
@@ -3281,3 +3281,35 @@ pub fn setsockopt(fd: fd_t, level: u32, optname: u32, opt: []const u8) SetSockOp
         else => |err| return unexpectedErrno(err),
     }
 }
+
+// TODO support for non-null terminated strings?
+pub fn memfd_createC(name: [*:0]const u8, flags: usize) !fd_t {
+    if (builtin.os != .linux) @compileError("memfd_create() not implemented for this target");
+
+    const rc = linux.memfd_create(name, flags);
```

Comment:
This may potentially have a libc wrapper. Use `system.memfd_create`?
Which makes me realise: could you add an `extern` definition for the libc version?